web/src/test/java/DtoValidationTest.java

```diff
@@ -0,0 +1,178 @@
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.stream;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Utility-Tester for matching DTOs (Data Transfer Objects) with their DOs (Domain Objects).
+ * Criteria for success includes such factors:
+ * <ul>
+ * <li>Matching number of fields: DTO has to have not more fields declared than the DO does.</li>
+ * <li>Matching naming of fields: DTO has to have same field naming as the DO does.</li>
+ * </ul>
+ * <p>
+ * Test created for using mapping models by convention.
+ * </p>
+ * This util stands not only for testing matching DTOs with their DOs,
+ * it also provides information about declared fields in tested Classes.
+ *
+ * @author Andrii Abramov on 13-May-16.
+ */
+public final class DtoValidationTest {
+
+    /**
+     * Base package to scan for DTO.
+     * Pattern for DTO Class should match "XXXDto" where XXX = DTO's DO name.
+     */
+    private static final String DTO_BASE_PACKAGE = "edu.softserve.zoo.dto";
+    /**
+     * Base package to scan for DO.
+     */
+    private static final String ENTITY_BASE_PACKAGE = "edu.softserve.zoo.model";
+    /**
+     * Stores the classes which potentially belong to DTO's base package but shouldn't be tested.
+     */
+    private static final List<Class<? extends BaseDto>> EXCLUDED_DTOS = Arrays.asList(BaseDto.class);
+
+    /**
+     * Stores pairs DTO against its DO.
+     */
+    private static Map<Class<?>, Class<?>> dtoToEntity;
+
+    /**
+     * Initializes the mapping between DTOs and their DOs.
+     * @throws IOException if any path is broken
+     * @throws URISyntaxException if any path is broken
+     */
+    @BeforeClass
+    public static void setUp() throws IOException, URISyntaxException {
+        dtoToEntity = extractDtos(DTO_BASE_PACKAGE)
+                .filter(e -> !EXCLUDED_DTOS.contains(e))
+                .collect(Collectors.toMap(
+                        Function.identity(),
+                        entity -> getEntityForDto(ENTITY_BASE_PACKAGE, entity)
+                ));
+
+        dtoToEntity.keySet()
+                .forEach(className -> System.out.println("Dto classes found to test: " + className.getSimpleName()));
+        dtoToEntity
+                .forEach((d, e) -> System.out.println("For " + d.getSimpleName() + " found " + e.getSimpleName())
+        );
+
+    }
+
+    /**
+     * Fetches all DTOs that match the specified pattern of Class name.
+     * @param basePackage package where DTOs are stored
+     * @return Stream of Class Objects initialized by convention if file names in basePackage
+     * @throws URISyntaxException
+     * @throws IOException
+     */
+    private static Stream<? extends Class<?>> extractDtos(String basePackage) throws URISyntaxException, IOException {
```

Comment:
use classutils instead. you just need to find successors of base DTO and base domain class and compare them